File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
import sys
def solution(keymap, targets):
    answer = []
    for target in targets:
        sum = 0
        for t in target:
            tmp = sys.maxsize
            for i in range(len(keymap)):
                if t in keymap[i]:
                    tmp= min(tmp, keymap[i].index(t) + 1)
            sum += tmp
            if tmp == sys.maxsize:
                answer.append(-1)
                break
        else:
            answer.append(sum)
    
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    
    static Map<Character, Integer> map = new HashMap();
    
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        initMap(keymap);

        
        for(int i = 0; i < targets.length; i++) {
            int sum = 0;
            boolean flag = false;
            for(int j = 0; j < targets[i].length(); j++) {
                if(map.containsKey(targets[i].charAt(j))){
                    sum += map.get(targets[i].charAt(j));
                } else{
                    answer[i] = -1;
                    flag = true;
                    break;
                }
            }
            if(flag)
                continue;
            answer[i] = sum;
        }
        return answer;
    }
    
    static void initMap(String[] keymap) {
        for(String s : keymap){
            for(int i = 0; i <s.length(); i++){
                if (map.containsKey(s.charAt(i))){
                    if(map.get(s.charAt(i)) > (i + 1)){
                        map.put(s.charAt(i), i + 1);
                    }
                } else{
                    map.put(s.charAt(i), i + 1);
                }
            }
        }
    }
}